ag_data/models.py

```diff
@@ -72,3 +72,39 @@ class AGMeasurement(models.Model):
 
 class AGActiveEvent(models.Model):
     agevent = models.ForeignKey(AGEvent, null=True, on_delete=models.SET_NULL)
+
+
+class ErrorLog(models.Model):
+    # declearation error
+    UNKNOWN_FMT = "UNKNOWN_FORMAT"
+    MISSING_COL = "MISSING_COLUMN"
+    MISSING_FIELD_IN_RAW = "MISSING_FIELD_IN_RAW_READING"
+    INVALID_COL_NM = "INVALID_COLUMN_NAME"
+    INVALID_COL_VL = "INVALID_COLUMN_VALUE"
+    INVALID_FIELD_IN_RAW = "INVALID_FIELD_IN_RAW_READING"
+    ERROR_F_PROC_MMT = "FORMULA_PROCESS_MEASUREMENT_ERROR"
+    EXTRA_KEYVAL_IN_MMT = "EXTRANEOUS_KEY_VALUE_PAIR_IN_MEASUREMENT"
+    OTHER = "OTHER_ERROR"
+
+    ERROR_CODE_CHOICES = [
+        (UNKNOWN_FMT, "Unknown Format"),
+        (MISSING_COL, "Missing Column"),
+        (MISSING_FIELD_IN_RAW, "Missing Field In Raw Reading"),
+        (INVALID_COL_NM, "Invalid Column Name"),
+        (INVALID_COL_VL, "Invalid Column Value"),
+        (INVALID_FIELD_IN_RAW, "Invalid Field In Raw Reading"),
+        (ERROR_F_PROC_MMT, "Error When Formula Processing Measurement"),
+        (EXTRA_KEYVAL_IN_MMT, "Extraneous Key-Value Pair In Measurement"),
+        (OTHER, "Other Error"),
+    ]
+
+    error_id = models.AutoField(primary_key=True)
```

Comment:
I think our other models are using UUID as keys. I feel it would work well for this model as well.

Reply:
In this case, both UUID and autofield ID work, I would say this doesn't matter unless UUID has more advantages than this original plan.

Reply:
yes, we don't want to mix and use both, unless there is some implications for that. the tables using numeric ids are sensor tables, which is due to compatibility with the hardware system. I'm okay if we keep this pr unchanged (to use numeric id).